rpc/src/router.rs

```diff
@@ -0,0 +1,153 @@
+use std::{collections::HashMap, marker::PhantomData};
+
+use serde::{Deserialize, Serialize};
+
+pub struct Router {
+    pub routes: HashMap<String, Box<dyn Handler + 'static + Send + Sync>>,
+}
+
+impl Router {
+    pub fn new() -> Self {
+        Router {
+            routes: HashMap::new(),
```

Comment:
Pub-sub pattern can be supported by adding a routing table here which is a HashMap that stores the topic and a Vec of callbacks - `HashMap<String, Vec<HandlerFunction>>`. And exposing `publish` and `subscribe`, `unsubscribe` that allow publishing data and modifying subscriptions.